### PR TITLE
Added UiState and basic animations #28

### PR DESCRIPTION
**Pull request by DreyerChris**

### CODE REVIEW

File: src/state/positioner.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+export const toUiState = (prevUiState: UiState | null, gameState: GameState): UiState => {
+  return {
+    cardPositions: calculateCardPositions(prevUiState, gameState),
+    deckPositions: calculateDeckPositions(gameState),
+  };
+};
+
+const calculateCardPositions = (prevUiState: UiState | null, gameState: GameState): GamePieceTransforms => {
+  return {
+    ...drawAnimalCards(prevUiState, gameState),
```
Comment (H1D) [inline src/state/positioner.ts:34]: Minor. Why `draw...`? maybe position/set/put/lay

File: src/state/types.ts [hunk 2]
```diff
@@ -109,3 +109,24 @@ export interface DisasterCard extends GamePieceBase {
...
+export interface Coordinate {
+  x: number;
+  y: number;
+  z: number;
+}
+
+export interface GamePieceTransform {
+  position: Coordinate;
+  initialPosition?: Coordinate;
+  rotation: Coordinate;
```
Comment (H1D) [inline src/state/types.ts:122]: we might need `initialRotation` as well

File: src/state/positioner.ts (hunk 1 above)
Comment (DreyerChris) [reply src/state/positioner.ts:34]: Yeah good point I called it draw initially and just kept the name, will think of something better

File: src/state/types.ts (hunk 2 above)
Comment (DreyerChris) [reply src/state/types.ts:122]: Yes I will add it

**Pull request merged**